clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -4424,12 +4427,21 @@ private QuorumInfo createQuorumResult(final DescribeQuorumResponseData.Partition
                     .map(this::translateReplicaState)
                     .collect(Collectors.toList());
 
+                List<QuorumInfo.Node> nodes = nodeCollection.stream().map(n -> {
+                    List<Endpoint> endpoints = n.listeners().stream()
+                        .map(l -> new Endpoint(l.name(), SecurityProtocol.forId(l.securityProtocol()), l.host(), l.port()))
```

Comment:
You should not be using Endpoint here since it's an internal class. Also we don't need/want to expose it in the admin API.

Reply:
Replaced with the `RaftVoterEndpoint`